File: dehaze-java/src/main/java/com/pei/dehaze/model/entity/AuditLog.java
package com.pei.dehaze.model.entity;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 业务操作审计日志（MongoDB 集合 {@code audit_log}）。
 *
 * <p>集合由 java 与 dehaze-python 共写共读，python 写入的文档键为 snake_case，
 * 故字段必须显式 {@link Field} 映射，否则同集合内两端数据互不可见。
 */
@Data
@Document(collection = "audit_log")
public class AuditLog implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;

    @Id
    private String id;

    @Field("operator_id")
    private Long operatorId;

    @Field("target_type")
    private String targetType;

    @Field("target_id")
    private Object targetId;

    private String action;

    private String module;

    @Field("before_value")
    private Object beforeValue;

    @Field("after_value")
    private Object afterValue;

    private String ip;

    @Field("user_agent")
    private String userAgent;

    @Indexed
    @Field("create_time")
    private LocalDateTime createTime;
}

File: dehaze-go/internal/model/audit_log.go
package model

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

// AuditLog 审计日志（MongoDB 集合 audit_log）。
//
// bson 键名一律 snake_case，与 python 写入端逐字一致（`mongo_audit_log_repository.create_audit`：
// operator_id/target_type/target_id/action/module/before_value/after_value/ip/user_agent/create_time）。
// 注意：这里必须是 snake_case —— 两边若不一致，go 写、python 读会静默全 None（反之亦然）；
// json 标签仍为 camelCase，供 API 出参使用。
type AuditLog struct {
	ID          primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	OperatorID  int64              `bson:"operator_id" json:"operatorId"`
	TargetType  string             `bson:"target_type" json:"targetType"`
	TargetID    interface{}        `bson:"target_id,omitempty" json:"targetId"`
	Action      string             `bson:"action" json:"action"`
	Module      string             `bson:"module" json:"module"`
	BeforeValue interface{}        `bson:"before_value,omitempty" json:"beforeValue"`
	AfterValue  interface{}        `bson:"after_value,omitempty" json:"afterValue"`
	IP          string             `bson:"ip" json:"ip"`
	UserAgent   string             `bson:"user_agent" json:"userAgent"`
	CreateTime  time.Time          `bson:"create_time" json:"createTime"`
}

func (AuditLog) CollectionName() string {
	return "audit_log"
}
